common/device/api/u_device.h

```diff
@@ -428,6 +428,18 @@ int32_t uDeviceOpen(const uDeviceCfg_t *pDeviceCfg,
  */
 int32_t uDeviceClose(uDeviceHandle_t devHandle, bool powerOff);
 
+/** Attach user context to device.
```

Comment:
Should mention that this is not thread-safe and hence should not be called at the same time as any other `uDevice` API function might be called, something like:

```
/** Attach user context to device.
 *
 * This is NOT thread-safe and should NOT be called when any other
 * uDevice API function might be called.  Best call it just after
 * calling uDeviceOpen() and before calling anything else.
 *
 * @param pUserContext   a user context to set.
 */
```